src/services/entity_repository.js

```diff
@@ -201,23 +213,35 @@ export default class EntityRepository {
       }
     };
 
-    await this.db.collection('assets').update(thisBundleQuery, update, {multi: true});
-    await this.db.collection('events').update(thisBundleQuery, update, {multi: true});
+    await this.db.collection('assets').updateMany(thisBundleQuery, update);
+    await this.db.collection('events').updateMany(thisBundleQuery, update);
   }
 
-  async getBundle(bundleId) {
-    return await this.db.collection('bundles').findOne({bundleId}, {fields: this.blacklistedFields});
+  async storeBundleShelteringExpirationDate(bundleId, expirationDate) {
+    await this.db.collection('bundles').updateOne({bundleId}, {
+      $set: {
+        'metadata.holdUntil': expirationDate
+      }
+    });
   }
 
-  hideEventDataIfNecessary(event, accessLevel) {
-    if (!event) {
-      return null;
-    }
-    return event.content.idData.accessLevel <= accessLevel ? event : pick(event, 'content.data');
+  async getExpiredBundleIds() {
+    const now = getTimestamp();
+    return this.db.collection('bundles').find({
+      $or: [
+        {'metadata.holdUntil': {$not: {$type: 'int'}}},
```

Comment:
You should consider putting and quering this from the `repository` dict instead of the `metadata` dict. The former is meant to be used for internal bookkeeping of the repository, and never leaves it's boundaries (it gets filtered out).